Build error trying to compile on Fedora/Arch with GCC 16.1.1

On `v2026.3.rc1` I'm seeing the following compilation error on Fedora and Arch Linux distros using GCC version 16.1.1

```
[ 44%] Building CXX object source/opt/CMakeFiles/SPIRV-Tools-opt.dir/pass_manager.cpp.o
In file included from /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:22:
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:310:9: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘unsigned char [384]’ [-Werror=array-bounds=]
  310 |     if (cpu_time_ >= 0 && Timer::CPUTime() >= 0)
      |         ^~~~~~~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:311:17: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘unsigned char [384]’ [-Werror=array-bounds=]
  311 |       cpu_time_ += Timer::CPUTime();
      |       ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:313:17: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘unsigned char [384]’ [-Werror=array-bounds=]
  313 |       cpu_time_ = -1;
      |       ~~~~~~~~~~^~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:315:9: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘unsigned char [384]’ [-Werror=array-bounds=]
  315 |     if (wall_time_ >= 0 && Timer::WallTime() >= 0)
      |         ^~~~~~~~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:316:18: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘unsigned char [384]’ [-Werror=array-bounds=]
  316 |       wall_time_ += Timer::WallTime();
      |       ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:318:18: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘unsigned char [384]’ [-Werror=array-bounds=]
  318 |       wall_time_ = -1;
      |       ~~~~~~~~~~~^~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:320:9: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘unsigned char [384]’ [-Werror=array-bounds=]
  320 |     if (usr_time_ >= 0 && Timer::UserTime() >= 0)
      |         ^~~~~~~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:321:17: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘unsigned char [384]’ [-Werror=array-bounds=]
  321 |       usr_time_ += Timer::UserTime();
      |       ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:323:17: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘unsigned char [384]’ [-Werror=array-bounds=]
  323 |       usr_time_ = -1;
      |       ~~~~~~~~~~^~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:325:9: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘unsigned char [384]’ [-Werror=array-bounds=]
  325 |     if (sys_time_ >= 0 && Timer::SystemTime() >= 0)
      |         ^~~~~~~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:326:17: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘unsigned char [384]’ [-Werror=array-bounds=]
  326 |       sys_time_ += Timer::SystemTime();
      |       ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:328:17: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘unsigned char [384]’ [-Werror=array-bounds=]
  328 |       sys_time_ = -1;
      |       ~~~~~~~~~~^~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:330:9: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘unsigned char [384]’ [-Werror=array-bounds=]
  330 |     if (rss_ >= 0 && Timer::RSS() >= 0)
      |         ^~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:331:12: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘unsigned char [384]’ [-Werror=array-bounds=]
  331 |       rss_ += Timer::RSS();
      |       ~~~~~^~~~~~~~~~~~~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:333:12: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘unsigned char [384]’ [-Werror=array-bounds=]
  333 |       rss_ = -1;
      |       ~~~~~^~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:335:9: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘unsigned char [384]’ [-Werror=array-bounds=]
  335 |     if (pgfaults_ >= 0 && Timer::PageFault() >= 0)
      |         ^~~~~~~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:336:17: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘unsigned char [384]’ [-Werror=array-bounds=]
  336 |       pgfaults_ += Timer::PageFault();
      |       ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:338:17: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘unsigned char [384]’ [-Werror=array-bounds=]
  338 |       pgfaults_ = -1;
      |       ~~~~~~~~~~^~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:310:9: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘void [384]’ [-Werror=array-bounds=]
  310 |     if (cpu_time_ >= 0 && Timer::CPUTime() >= 0)
      |         ^~~~~~~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:311:17: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘void [384]’ [-Werror=array-bounds=]
  311 |       cpu_time_ += Timer::CPUTime();
      |       ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:313:17: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘void [384]’ [-Werror=array-bounds=]
  313 |       cpu_time_ = -1;
      |       ~~~~~~~~~~^~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:315:9: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘void [384]’ [-Werror=array-bounds=]
  315 |     if (wall_time_ >= 0 && Timer::WallTime() >= 0)
      |         ^~~~~~~~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:316:18: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘void [384]’ [-Werror=array-bounds=]
  316 |       wall_time_ += Timer::WallTime();
      |       ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:318:18: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘void [384]’ [-Werror=array-bounds=]
  318 |       wall_time_ = -1;
      |       ~~~~~~~~~~~^~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:320:9: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘void [384]’ [-Werror=array-bounds=]
  320 |     if (usr_time_ >= 0 && Timer::UserTime() >= 0)
      |         ^~~~~~~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:321:17: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘void [384]’ [-Werror=array-bounds=]
  321 |       usr_time_ += Timer::UserTime();
      |       ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:323:17: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘void [384]’ [-Werror=array-bounds=]
  323 |       usr_time_ = -1;
      |       ~~~~~~~~~~^~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:325:9: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘void [384]’ [-Werror=array-bounds=]
  325 |     if (sys_time_ >= 0 && Timer::SystemTime() >= 0)
      |         ^~~~~~~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:326:17: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘void [384]’ [-Werror=array-bounds=]
  326 |       sys_time_ += Timer::SystemTime();
      |       ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:328:17: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘void [384]’ [-Werror=array-bounds=]
  328 |       sys_time_ = -1;
      |       ~~~~~~~~~~^~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:330:9: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘void [384]’ [-Werror=array-bounds=]
  330 |     if (rss_ >= 0 && Timer::RSS() >= 0)
      |         ^~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:331:12: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘void [384]’ [-Werror=array-bounds=]
  331 |       rss_ += Timer::RSS();
      |       ~~~~~^~~~~~~~~~~~~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:333:12: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘void [384]’ [-Werror=array-bounds=]
  333 |       rss_ = -1;
      |       ~~~~~^~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:335:9: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘void [384]’ [-Werror=array-bounds=]
  335 |     if (pgfaults_ >= 0 && Timer::PageFault() >= 0)
      |         ^~~~~~~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:336:17: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘void [384]’ [-Werror=array-bounds=]
  336 |       pgfaults_ += Timer::PageFault();
      |       ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
In member function ‘virtual void spvtools::utils::CumulativeTimer::Stop()’,
    inlined from ‘spvtools::utils::ScopedTimer<TimerType>::~ScopedTimer() [with TimerType = spvtools::utils::Timer]’ at /sdk/source/SPIRV-Tools/source/util/timer.h:260:16,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:78:3:
/sdk/source/SPIRV-Tools/source/util/timer.h:338:17: error: array subscript ‘spvtools::utils::CumulativeTimer[0]’ is partly outside array bounds of ‘void [384]’ [-Werror=array-bounds=]
  338 |       pgfaults_ = -1;
      |       ~~~~~~~~~~^~~~
In constructor ‘spvtools::utils::ScopedTimer<TimerType>::ScopedTimer(std::ostream*, const char*, bool) [with TimerType = spvtools::utils::Timer]’,
    inlined from ‘spvtools::opt::Pass::Status spvtools::opt::PassManager::Run(spvtools::opt::IRContext*)’ at /sdk/source/SPIRV-Tools/source/opt/pass_manager.cpp:57:5:
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
  253 |       : timer(new TimerType(out, measure_mem_usage)), tag_(tag) {
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/sdk/source/SPIRV-Tools/source/util/timer.h:253:15: note: object of size 384 allocated by ‘operator new’
cc1plus: all warnings being treated as errors
make[2]: *** [source/opt/CMakeFiles/SPIRV-Tools-opt.dir/build.make:1241: source/opt/CMakeFiles/SPIRV-Tools-opt.dir/pass_manager.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:798: source/opt/CMakeFiles/SPIRV-Tools-opt.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

--- a/source/util/timer.h
+++ b/source/util/timer.h
@@ -23,6 +23,11 @@
 #include <cassert>
 #include <iostream>
 
+#if defined(__GNUG__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Warray-bounds"
+#endif
+
 // A macro to call spvtools::utils::PrintTimerDescription(std::ostream*, bool).
 // The first argument must be given as std::ostream*. If it is NULL, the
 // function does nothing. Otherwise, it prints resource types measured by Timer
@@ -381,6 +386,10 @@ class CumulativeTimer : public Timer {
 
 }  // namespace utils
 }  // namespace spvtools
+
+#if defined(__GNUG__)
+#pragma GCC diagnostic pop
+#endif
 
 #else  // defined(SPIRV_TIMER_ENABLED)
 

